lib/wsfed.js

```diff
@@ -79,6 +79,10 @@ module.exports = function(options) {
       var profileMap = options.profileMapper(user);
       var claims = profileMap.getClaims(options);
       var ni = profileMap.getNameIdentifier(options);
+      if (!ni || !ni.nameIdentifier) {
```

Comment:
Good question @luuuis! I feel like the responsibility of the middleware should be to guarantee that meaningful response is returned. Otherwise, it's going to return the empty `nameIdentifier` without alert to the client and the client has to deal with it somehow. While empty `nameIdentifier` be a clear indication of the problem within used profile mapper.

On the other hand, somebody might say that it's the job of the profile mapper to guarantee the existence of `nameIdentifier` and fail otherwise, but it's not the obvious assumption. And I feel like failing fast is more robust and predictable option. WDYT?

Reply:
That makes sense. Even though the `<NameIdentifier>` is optional, for our purposes we always want to include one in the SAML assertions so it is appropriate to fail fast.

Reply:
Agreed, for our purposes this would work just fine. I'm thinking: could this potentially be a breaking change for consumers of this lib? In other words, do you know if the name identifier is strictly required in the assertion and hence not finding one should result in us throwing?

Reply:
Hm, it's a good question. So it might be a breaking change for those consumers, that:
* use custom profile mapper that returns name identifier object without `nameIdentifier` claim inside
* used profile mapper (default or custom) maps empty name identifier
and expect an assertion to be successfully formed. 

Reading [SAML v1.1 spec](https://www.oasis-open.org/committees/download.php/3406/oasis-sstc-saml-core-1.1.pdf), I don't see that `<NameIdentifier>` has to be a non-empty string. So it might be a valid case for some flow?

On the other hand, since a similar change has been done in [samlp](https://github.com/auth0/node-samlp/commit/a34449931dbdc688b6ed85567919328036fa90aa) it seems nobody explicitly requested a possibility to support this scenario. What if this change will be released as MINOR or MAJOR change?

Reply:
After talking with @FadyMak, given that we don't know if any consumer of the library really needs to explicitly pass empty `nameIdentifier`,  it was decided not to introduce an extra complexity (of supporting `requireNameIdentifier` config option) for now and to go with this breaking change, releasing a major version. 

This new config option `requireNameIdentifier` might be added in the future if the corresponding feedback will be received.